function.go

```diff
@@ -0,0 +1,126 @@
+package currency
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+// validateISOCodeAlpha : returns a formatted ISO alpha code or an error if the ISO is not found
+func validateISOCodeAlpha(alpha string) (string, error) {
+	alpha = strings.ToUpper(alpha)
+	for key := range CurrencyList {
+		if key == alpha {
+			return alpha, nil
+		}
+	}
+	return "", errorInvalidISO
+}
+
+// validateISOCodeNumeric : returns a formatted ISO numeric code or an error if the ISO is not found
+func validateISOCodeNumeric(num string) (string, error) {
+	for _, value := range CurrencyList {
+		if value.Numeric == num {
+			return value.Numeric, nil
+		}
+	}
+	return "", errorInvalidISO
+}
+
+// getISOFromAlpha : returns the Currency struct for that ISO code alpha
+func getISOFromAlpha(alpha string) (Currency, error) {
+	alpha, err := validateISOCodeAlpha(alpha)
+	if err != nil {
+		return Currency{}, err
+	}
+	output := CurrencyList[alpha]
+	return output, nil
+}
+
+// getAlphaFromISOCodeNumeric : returns a formatted ISO alpha code from the ISO numeric counterpart
+func getAlphaFromISOCodeNumeric(num string) (string, error) {
+	for _, value := range CurrencyList {
+		if value.Numeric == num {
+			return value.Alpha, nil
+		}
+	}
+	return "", errorInvalidISO
+}
+
+// convertToString : returns the uint number as a string
+func convertToString(num uint, exp int) string {
+	floatNum := float64(num)
+	newNum := float64(floatNum / math.Pow10(exp))
+	output := fmt.Sprintf("%.*f", exp, newNum)
+	return output
+}
+
+// splitString : returns a map of strings separated by decimal place
+func splitString(str string) []string {
+	return strings.Split(str, ".")
+}
+
+// reverseString : returns a reversed string for delimiter formatting
+func reverseString(str string) string {
+	var output string
+	for key := len(str) - 1; key >= 0; key-- {
+		output += string(str[key])
+	}
+	return output
+}
+
+// insertDelimiter : returns a new string with delimiter formatting
+func insertDelimiter(str string, sep int, del string) string {
+	var output strings.Builder
+	for key, val := range str {
+		if key%sep == 0 && key != 0 {
+			fmt.Fprintf(&output, del+string(val))
```

Comment:
Builder has a write function. output.WriteString("Thing you want to add to the string")